examples/pytorch/gxn/networks.py

```diff
@@ -0,0 +1,305 @@
+from typing import List, Tuple, Union
+
+from layers import *
+import torch
+import torch.nn
+import torch.nn.functional as F
+import dgl.function as fn
+from dgl.nn.pytorch.glob import SortPooling
+
+
+class GraphCrossModule(torch.nn.Module):
+    """
+    Description
+    -----------
+    The Graph Cross Module used by Graph Cross Networks.
+    This module only contains graph cross layers.
+
+    Parameters
+    ----------
+    pool_ratios : Union[float, List[float]]
+        The pooling ratios (for keeping nodes) for each layer.
+        For example, if `pool_ratio=0.8`, 80\% nodes will be preserved.
+        If a single float number is given, all pooling layers will have the
+        same pooling ratio.
+    in_dim : int
+        The number of input node feature channels.
+    out_dim : int
+        The number of output node feature channels.
+    hidden_dim : int
+        The number of hidden node feature channels.
+    cross_weight : float, optional
+        The weight parameter used in graph cross layers
+        Default: :obj:`1.0`
+    fuse_weight : float, optional
+        The weight parameter used at the end of GXN for channel fusion.
+        Default: :obj:`1.0`
+    """
+    def __init__(self, pool_ratios:Union[float, List[float]], in_dim:int,
+                 out_dim:int, hidden_dim:int, cross_weight:float=1.,
+                 fuse_weight:float=1., dist:int=1, num_cross_layers:int=2):
+        super(GraphCrossModule, self).__init__()
+        if isinstance(pool_ratios, float):
+            pool_ratios = (pool_ratios, pool_ratios)
+        self.cross_weight = cross_weight
+        self.fuse_weight = fuse_weight
+        self.num_cross_layers = num_cross_layers
+
+        # build network
+        self.start_gcn_scale1 = GraphConvWithDropout(in_dim, hidden_dim)
+        self.start_gcn_scale2 = GraphConvWithDropout(hidden_dim, hidden_dim)
+        self.end_gcn = GraphConvWithDropout(2 * hidden_dim, out_dim)
+
+        self.index_select_scale1 = IndexSelect(pool_ratios[0], hidden_dim, act="prelu", dist=dist)
+        self.index_select_scale2 = IndexSelect(pool_ratios[1], hidden_dim, act="prelu", dist=dist)
+        self.start_pool_s12 = GraphPool(hidden_dim)
```

Comment:
Is it possible to use loop and lists for a more flexible implementation? In that case it is easy to control the number of pool/unpool operations.

Reply:
Actually, the author only uses models with the number of graph cross layers less than 3. Also, there is some difference between these layers. For example line 121 and line 140 in this file.

Reply:
That seems to me is just adding skip connections with regard to `num_cross_layers`, and can still be generalized. But anyway, if no one's requesting a flexible version, we can keep it as is for now.